### DISCUSSION TITLE
trying to get reinforcement with specific mp example 81 which is cheaper

### DISCUSSION THREAD

**Discussion by MaroYqr** [coccoinomane/crabada.py#96] [Q&A] [answered]
hi ,

newb here in python

trying to modif the def query so as to return with crab that has 81 mp

def query(self, game: Game) -> dict[str, Any]:
        return {
            "limit": 100,	
            "mine_point" : 81,
            "orderBy": "price",
            "order": "asc",
        }

is this the right way to do it? currently returning no crabs that meets requirement.

**Comment by yigitest** [comment, marked as answer]
You could try lowering `USER_1_REINFORCEMENT_MAX_PRICE` value in the .env file.

**Reply by MaroYqr** [reply]
I tried it with the highestcostMp strategy with reinforcement max at 10 . Doesnt return a crab for me

**Reply by yigitest** [reply]
Currently, there are 4192 crabs in the tavern but we can only fetch 100 of them at a time (Pagination feature is not implemented yet). So finding  MP=81 crabs with cheaper TUS costs is might not be possible for every time. 

But you can filter by `class_ids` instead of `mine_point`. For example prime's id is 3:

 ```
 return {
"limit": 100,
"class_ids[]" : 3,
"orderBy": "time_point",
"order": "desc",
}
```

You can try different parameters and see the results yourself here: https://idle-api.crabada.com/public/idle/crabadas/lending?class_ids[]=3&orderBy=time_point&order=desc&page=1&limit=100

**Reply by MaroYqr** [reply]
This works 👍  . Thanks for the Help .

**Reply by CBSInvest** [reply]
@MaroYqr 
Did u replace the minepoint in the coding with the Class_ids to solve the error msg getting some reinforcement crabs.
Do u know the code for Craboid ?

@yigitest 
is it restrice to 100 from the devs of Crabada that we are only possible to fetch 100?

**Reply by MaroYqr** [reply]
I did not change the code @yigitest provided , it works , what i did is create a new strategy by replacing the code in HighestcostMP.py and importing the new strategy in file ReinforceStrategyFactory.py. 

The Class_ids for crabtoid is 5

**Reply by CBSInvest** [reply]
@MaroYqr 
Awesome thx, did the same now with HighestcostBP for the looting so lets see if it works - 
Couldnt find the list for the Class_ids but i think Surge is 1 and Bulk is 4 correct !

**Reply by zishanali06** [reply]
> Currently, there are 4192 crabs in the tavern but we can only fetch 100 of them at a time (Pagination feature is not implemented yet). So finding MP=81 crabs with cheaper TUS costs is might not be possible for every time.
> 
> But you can filter by `class_ids` instead of `mine_point`. For example prime's id is 3:
> 
> ```
> return {
> "limit": 100,
> "class_ids[]" : 3,
> "orderBy": "time_point",
> "order": "desc",
> }
> ```
> 
> You can try different parameters and see the results yourself here: [https://idle-api.crabada.com/public/idle/crabadas/lending?class_ids[]=3&orderBy=time_point&order=desc&page=1&limit=100](https://idle-api.crabada.com/public/idle/crabadas/lending?class_ids%5B%5D=3&orderBy=time_point&order=desc&page=1&limit=100)

Hey you know how you got this URL above. What is the endpoint URL for if I want to see which loots are open(is this possible)? For on the swimmer network? Trying to do something and need that URL.
